backend/prompt_studio/prompt_studio_index_manager/prompt_studio_index_helper.py

```diff
@@ -24,32 +24,29 @@ def handle_index_manager(
         if is_summary:
             index_id = "summarize_index_id"
 
-        try:
-            index_manager: Optional[IndexManager] = IndexManager.objects.get(
-                document_manager=document, profile_manager=profile_manager
-            )
-        except IndexManager.DoesNotExist:
-            index_manager = None
+        args: dict[str, str] = dict()
+        args["document_manager"] = document
+        args["profile_manager"] = profile_manager
+        args[index_id] = doc_id
 
-        args: dict[str, str] = {f"{index_id}": doc_id}
+        try:
+            # Create or get the existing record for this document and
+            # profile combo
+            index_manager, _ = IndexManager.objects.get_or_create(**args)
 
-        index_ids_list = []
-        if index_manager:
             index_ids = index_manager.index_ids_history
             index_ids_list = json.loads(index_ids) if index_ids else []
             if doc_id not in index_ids:
                 index_ids_list.append(doc_id)
-        else:
-            index_ids_list.append(doc_id)
 
-        args["index_ids_history"] = json.dumps(index_ids_list)
+            args["index_ids_history"] = json.dumps(index_ids_list)
 
-        if index_manager:
+            # Update the record with the index id
             result: IndexManager = IndexManager.objects.filter(
                 index_manager_id=index_manager.index_manager_id
             ).update(**args)
-        else:
-            args["document_manager"] = document
-            args["profile_manager"] = profile_manager
-            result = IndexManager.objects.create(**args)
+
+        except Exception as e:
```

Comment:
Why do we require a catch all here? If we make future updates to this try - catch block, it might suppress the original exception and always return this message instead

Reply:
This it to catch any othe system/db related errors and make sure that exception is not shown to user. The caller of this method only handles IndexingError exception for now.

Reply:
This it to catch any other system/db related errors and make sure that exception is not shown to user. The caller of this method only handles IndexingError exception for now.
What is frequent updates?

Reply:
@gaya3-zipstack does this `raise from e` show us the original exception?